runners/gradle-plugin/src/main/kotlin/org/jetbrains/dokka/gradle/GradleExternalDocumentationLinkBuilder.kt

```diff
@@ -66,10 +67,10 @@ class GradleExternalDocumentationLinkBuilder(
      */
     @Optional
     @Input
-    val packageListUrl: Property<URL?> = project.objects.safeProperty()
+    val packageListUrl: Property<URL> = project.objects.property()
```

Comment:
I wonder what the difference is between `Property<URL>` (now) and `Property<URL?>` (before)

It seems that the property with `Property<URL>` is still optional, nothing fails if I set `null`. How do you then determine if something is nullable or not? 

Trying to understand the logic for calling `.get()` vs `.orNull` and whether it's possible that this change can introduce regression where a previously optional (nullable) property becomes required because `get()`  throws an error on `null`. 

I'm trying to break something locally, but it doesn't want to fail no matter what I do, and I don't understand why :sweat_smile:

Reply:
I think that since `Property<T>` is defined in Java, and it's not annotated as non-null, `T` is a platform type so it's always considered nullable.

My guess is that it's supposed to be used in tandem with the custom helper functions (like `safeConvention()`), which are supposed to 'extend' the existing Gradle API to try and make it impossible to set `null` into a `Property<String>`, but I'm not sure.